processor/src/main/java/org/mapstruct/ap/internal/processor/DefaultModelElementProcessorContext.java

```diff
@@ -143,10 +143,15 @@ public void printMessage(Element e, Message msg, Object... args) {
 
         @Override
         public void printMessage(Element e, AnnotationMirror a, Message msg, Object... args) {
-            String message = String.format( msg.getDescription(), args );
-            delegate.printMessage( msg.getDiagnosticKind(), message, e, a );
-            if ( msg.getDiagnosticKind() == Kind.ERROR ) {
-                isErroneous = true;
+            if ( a == null ) {
```

Comment:
Shouldn't we just call the correct method here instead of doing these checks. I thought that the idea behind our `FormattingMessager` was to just delegate to the same method